user-console/src/main/java/org/pentaho/mantle/client/ui/BurgerMenuPopup.java

```diff
@@ -0,0 +1,116 @@
+ * See the GNU Lesser General Public License for more details.
+ *
+ * Copyright (c) 2023 Hitachi Vantara. All rights reserved.
+ */
+
+package org.pentaho.mantle.client.ui;
+
+import com.google.gwt.dom.client.Element;
+import com.google.gwt.dom.client.EventTarget;
+import com.google.gwt.dom.client.NativeEvent;
+import com.google.gwt.event.dom.client.KeyCodes;
+import com.google.gwt.user.client.Event;
+import com.google.gwt.user.client.ui.DecoratedPopupPanel;
+import com.google.gwt.user.client.ui.Focusable;
+import com.google.gwt.user.client.ui.Widget;
+
+public class BurgerMenuPopup extends DecoratedPopupPanel {
+  private Widget menuButton;
+
+  public BurgerMenuPopup( Widget menuButton, BurgerMenuBar menuBar ) {
+    super( true );
+
+    // Imitate BurgerMenuBar's Popup Panel classes
+    setStyleName( "gwt-MenuBarPopup" );
+    addStyleName( "pen-BurgerMenuBarPopup" );
+
+    this.menuButton = menuButton;
+
+    setWidget( menuBar );
+
+    // Avoid inheriting CSS styling for .mainMenubar.
+    menuBar.getElement().setId( "burgerMenubar" );
+
+    menuButton.getElement().setAttribute( "aria-controls", menuBar.getElement().getId() );
+    menuButton.getElement().setAttribute( "aria-expanded", "true" );
+
+    // Mouse-down'ing on the button should not trigger the default autoHide behavior, of hiding the PopupPanel.
+    // This would hide the popup, however, the click event on the button would then immediately re-open the menu popup.
+    // Instead, the dialog is not closed when clicking on the button and then it is hidden instead in the
+    // click preview event, in which the event is also canceled, to prevent the default action of opening the popup...
+    addAutoHidePartner( menuButton.getElement() );
+
+    addCloseHandler( ev -> {
+      menuButton.getElement().removeAttribute( "aria-controls" );
+      menuButton.getElement().removeAttribute( "aria-expanded" );
+
+      if ( menuButton instanceof Focusable ) {
+        ( (Focusable) menuButton ).setFocus( true );
+      }
+    } );
+  }
+
+  @Override
+  protected void onPreviewNativeEvent( Event.NativePreviewEvent event ) {
+    switch ( event.getTypeInt() ) {
+      case Event.ONKEYDOWN: {
+        switch ( event.getNativeEvent().getKeyCode() ) {
+          // Must handle ESCAPE for the top-level menu.
+          case KeyCodes.KEY_ESCAPE:
+          case KeyCodes.KEY_TAB: {
+            hide();
+            event.cancel();
+            return;
+          }
+          default:
+        }
+        break;
+      }
+
+      case Event.ONCLICK: {
+        // See comment above on constructor, near addAutoHidePartner.
+        if ( !eventTargetsPopupLocal( event.getNativeEvent() ) ) {
+          hide();
+          event.cancel();
+        }
+        break;
+      }
+      default:
+    }
+
+    super.onPreviewNativeEvent( event );
+  }
+
+  public void showMenu() {
+    setPopupPositionAndShow( ( offsetWidth, offsetHeight ) ->
```

Comment:
why are you creating an lambda with parameters  ` ( offsetWidth, offsetHeight )` and then not using them?

CustomDropDown#onBrowserEvent(event), creates a PositionCallback without arguments.

```
        popup.setPopupPositionAndShow( new PositionCallback() {
          public void setPosition( int offsetWidth, int offsetHeight ) {
            popup.setPopupPosition( getAbsoluteLeft(), getAbsoluteTop() + getOffsetHeight() - 1 );
          }
        } );
```

https://github.com/pentaho/pentaho-platform/blob/9.5.0.0/user-console/src/main/java/org/pentaho/mantle/client/ui/CustomDropDown.java#L186C45-L190

Reply:
Perhaps I don't understand, but doesn't this `setPosition( int offsetWidth, int offsetHeight )` have the same parameters that aren't being used?

Reply:
Never mind. You are correct it's the same thing. I was thrown off by not using the parameters.